File: 0056-merge-intervals/0056-merge-intervals.js
/**
 * @param {number[][]} intervals
 * @return {number[][]}
 */
var merge = function(intervals) {
    intervals.sort((a,b) => a[0]-b[0])
    const result = [intervals[0]]

    for(const int of intervals){
        const lastElementEndTime = result[result.length-1][1]
        if(int[0] <= lastElementEndTime){
            result[result.length-1][1] = Math.max(int[1], lastElementEndTime)
        } else {
            result.push(int)
        }
    }

    return result
};

File: 0056-merge-intervals/0056-merge-intervals.py
class Solution:
    def merge(self, intervals: List[List[int]]) -> List[List[int]]:
        merged = []

        intervals.sort(key=lambda x: (x[0], x[1]))

        for interval in intervals:
            if merged and interval[0] <= merged[-1][1]:
                merged[-1][1] = max(interval[1], merged[-1][1])
            else:
                merged.append(interval)
        
        return merged
        